graphics_manager/graphics_manager.py

```diff
@@ -16,40 +16,98 @@ def __init__(self) -> None:
         self.textures: dict = {}
         self.background: pg.Surface = None
         self.rects_to_update = []
+
+        self.canvas_game: pg.Surface = None
+        self.canvas_gui: pg.Surface = None
         logging.debug("GraphicsManager class initializer ran succesfully.")
 
-    def draw_group(self, group: pg.sprite.RenderUpdates) -> None:
+    def draw_object(self, object: GameObject, game:bool, background=None) -> None:
         """ Draws object.image texture on object.rect place."""
         logging.debug(" draw_object method of GraphicsManager called succesfully.")
-        group.clear(self.screen, self.background)
-        self.rects_to_update += group.draw(self.screen)
+        if background is None:
+            background = self.background
+        if game:
+            surface = self.canvas_game
+        else:
+            surface = self.canvas_gui
+
+        if game:
+            self.rects_to_update.append(surface.blit(object.image, (object.rect.x, object.rect.y)))
+        else:
+            self.rects_to_update.append(surface.blit(object.image, object.rect))
+
+        logging.debug("Object drawn succesfully.")
+
+    def draw_group(self, group: pg.sprite.RenderUpdates, game:bool, background=None) -> None:
+        """ Draws object.image texture on object.rect place."""
+        logging.debug(" draw_group method of GraphicsManager called succesfully.")
+        if background is None:
+            background = self.background
+        if game:
+            surface = self.canvas_game
+        else:
+            surface = self.canvas_gui
+        
+        group.clear(surface, background)
+
+        # Move rects down because of GUI
+        if game:
+            self.rects_to_update += [pg.Rect(rect.x, rect.y + Window.GUI_HEIGHT,
+                                             rect.width, rect.height) for rect in group.draw(surface)]
+        else:
+            self.rects_to_update += group.draw(surface)
+
+        logging.debug("Group drawn succesfully.")
+
+    def draw_rect(self, rect: pg.Rect, color: tuple, game:bool, background=None, width=Window.CARD_RECT_WIDTH) -> None:
+        """ Draws rect on screen. """
+        logging.debug(" draw_rect method of GraphicsManager called succesfully.")
+        if background is None:
+            background = self.background
+        if game:
+            surface = self.canvas_game
+        else:
+            surface = self.canvas_gui
+
+        pg.draw.rect(surface, color, rect, width)
+        self.rects_to_update.append(rect)
+
+        logging.debug("Rect drawn succesfully.")
+
 
     def load_all_textures(self) -> None:
         """Initializates load_all_textures method."""
         logging.debug("Load_all_functions called.")
-        self.textures = texture_loader.load_all_textures()
+        self.textures = TextureLoader.load_all_textures(self)
```

Comment:
CO JE SAKRA TOHLE?!?!??!?!?!?!?!!?!?!?!?!!??!?!?!??!? Upřímně netuším, jak to může fungovat, ale prosím nedělej to.

Reply:
Co je na tom spatne?

Reply:
já nevím Adame, ale připadá mi to docela jako kód. problém?

Reply:
TextureLoader není statická třída. Ten řádek nad tím by měl fungovat. Zjisti, proč ne.